### PR TITLE
added iOS build runners

### PR DESCRIPTION
**Pull request by JulianMHParadoxCat**
This pull request adds Ramses iOS compatibility (without renderer) build runners for both iOS Devices and iOS Simulators (which do not run unit tests, as this is not supported for iOS as of now).

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -79,7 +86,14 @@ jobs:
       working-directory: ${{runner.workspace}}/build
       run: |
         cmake $GITHUB_WORKSPACE -DCMAKE_BUILD_TYPE=${{ matrix.type }}
-      if: ${{ matrix.os == 'macos-12' }}
+      if: ${{ matrix.os == 'macos-12' && matrix.target-platform != 'iphoneos' && matrix.target-platform != 'iphonesimulator' }}
```
Comment (violinyanev) [inline .github/workflows/test.yml:89]: why not check for target-platform == host-os?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -90,3 +104,4 @@ jobs:
       working-directory: ${{runner.workspace}}/build
       shell: bash
       run: ctest -C ${{ matrix.type }} --exclude-regex '.*RNDSANDWICHTEST'
+      if: ${{ matrix.os != 'macos-12' && matrix.target-platform != 'iphoneos' && matrix.target-platform != 'iphonesimulator' }}
```
Comment (violinyanev) [inline .github/workflows/test.yml:107]: Shouldn't this be rather:
${{ matrix.os != 'macos-12' || (matrix.target-platform == 'host-os' }}

And maybe add a comment on top that tests can't be executed on iOS builds to clarify the complex condition a bit

**Review by violinyanev — CHANGES_REQUESTED** [review]

**Review by violinyanev — APPROVED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (JulianMHParadoxCat) [reply .github/workflows/test.yml]: Changed.

File: .github/workflows/test.yml (hunk 2 above)
Comment (JulianMHParadoxCat) [reply .github/workflows/test.yml]: Done.

**Pull request merged**